src/librustdoc/json/mod.rs

```diff
@@ -292,17 +292,16 @@ impl<'tcx> FormatRenderer<'tcx> for JsonRenderer<'tcx> {
             crate_version: self.cache.crate_version.clone(),
             includes_private: self.cache.document_private,
             index: index.into_iter().collect(),
-            paths: self
+            external_index: self
                 .cache
-                .paths
+                .external_paths
                 .iter()
-                .chain(&self.cache.external_paths)
                 .map(|(&k, &(ref path, kind))| {
                     (
                         from_item_id(k.into(), self.tcx),
-                        types::ItemSummary {
+                        types::ExternalItem {
                             crate_id: k.krate.as_u32(),
-                            path: path.iter().map(|s| s.to_string()).collect(),
+                            name: path.last().map(|s| s.to_string()),
```

Comment:
Only including external items sounds good to me, but can we still keep the path as before please?

I have just realized that I can use that as a workaround for https://github.com/rust-lang/rust/issues/99513

Reply:
Ok, I am starting to see why it is "questionably useful" now. `String` for example ends up having the path `alloc::string::String`, which is not really presentable. So feel free to keep this change as is; I don't think I can make use of the paths in their current form.

Sorry for the noise.

Reply:
> Only including external items sounds good to me, but can we still keep the `path` as before please?

This is the sole reason of this PR the rest is just improvements based on this. As I stated the problem with `path` are multiple (in fact even the name `path` is problematic because there isn't **a*** path to something but many different one possible for every items and I think we shouldn't lead to a "prefered one") and leaving it as is is not helpful to anyone (especially considering that there are discussion about stabilizing the all rustdoc-json output format).

> I have just now realized that I can use that as a workaround for #99513

This should have a proper fix, and should not be a reason to not remove `path`.